### PR TITLE
deps: Update thema, use CUE fork

### PR DESCRIPTION
**Pull request by sdboyer**
**What is this feature?**

Update thema to include the newly rewritten OpenAPI generator - grafana/thema#90

It also makes the necessary update to use the fork of CUE. See grafana/thema#91

### CODE REVIEW

**Comment by sdboyer** [comment]
So, i didn't run `make gen-cue`. When i did, i discovered that the old problem in CUE stdlib with incorrectly encoding OpenAPI (leading to all generated map types having `interface{}` values) had been resolved. Cool!

...Except that that meant the issue with openapi-codegen [generating a whole bunch of helper boilerplate](https://github.com/grafana/grafana/pull/56263#issuecomment-1270023581) we don't want or need was suddenly, newly, a problem.

So, i wrote a new AST manipulator that sucks out all those functions and the superfluous type with an `AdditionalProperties` map, sucks the type out of that map, inlines it back into any referring structs, and removes the superfluous type and extra funcs. Woohoo!

**Review by joanlopez — APPROVED** [review]
LGTM!

**Review by spinillos — APPROVED** [review]
It only updates the Go file, but we need to update Thema to re-generate TS files right?

I needed to check https://github.com/grafana/grafana/pull/56263#issuecomment-1270023581 at the same time that I was reading the PR to "understand" what you did here 😂. Ands looks like good.

**Review by marefr — APPROVED** [review]
LGTM

**Comment by sdboyer** [comment]
> I needed to check https://github.com/grafana/grafana/pull/56263#issuecomment-1270023581 at the same time that I was reading the PR to "understand" what you did here 😂. Ands looks like good.

Yeah it's weird - you can't see what the code actually does because all it does is _remove_ a bunch of code we don't want :) But, for the curious, it turns [this](https://gist.github.com/sdboyer/21015d38a90634a895acb0d712913eed#file-team_types_gen-go) into what's in this PR.

**Comment by sdboyer** [comment]
i'll update thema again to grab the latest cuetsy, so we roll it all together

**Comment by sdboyer** [comment]
@spinillos hmm, seems like the latest cuetsy has a problem we didn't catch in our tests

File: packages/grafana-schema/src/raw/team/x/team_types.gen.ts [hunk 1]
```diff
@@ -9,16 +9,16 @@
 // Run 'make gen-cue' from repository root to regenerate.
 
 export enum Permission {
-  Admin = 4,
-  Editor = 2,
-  Viewer = 1,
+  admin = 4,
+  editor = 2,
+  viewer = 1,
```
Comment (Clarity-89) [inline packages/grafana-schema/src/raw/team/x/team_types.gen.ts:14]: Wonder why this has been changed? We mostly use capitalized keys for enums in our codebase.

Comment (sdboyer) [reply packages/grafana-schema/src/raw/team/x/team_types.gen.ts]: They're explicitly written that way [in the kind decl](https://github.com/grafana/grafana/blob/0485cf34cda5ed6e209beabf70a80ab67f93bf42/kinds/structured/team/team_kind.cue#L30). If we want them uppercase, they need to be fixed in the input

File: go.mod [hunk 2]
```diff
@@ -60,7 +60,7 @@ require (
 	github.com/google/wire v0.5.0
 	github.com/gorilla/websocket v1.5.0
 	github.com/grafana/alerting v0.0.0-20221219210434-60ecaff51745
-	github.com/grafana/cuetsy v0.1.3-0.20221221204204-a320b05f3196
+	github.com/grafana/cuetsy v0.1.1
```
Comment (Clarity-89) [inline go.mod:63]: So we're excluding the latest cuetsy fixes from this PR?

**Comment by sdboyer** [comment]
i pulled out the cuetsy changes because they require more careful evaluation, and i wanted to deal with them in a separate PR

**Comment by sdboyer** [comment]
With this in, focusing on the cuetsy changes independently should be easier. i ultimately pulled it out because i saw a fair bit of change in terms of the use of `unknown` vs. `any`. It's my mistake, i shouldn't have let scope expand here in the first place - we have to start shifting into a stability mentality for generated code quite soon.

@spinillos could you please open a separate PR to pull in those cuetsy changes, and @Clarity-89 advise on impact to typescript? We can worry about updating cuetsy in thema later, updating it here is sufficient.

**Pull request merged**